manifests/xdr_credentials_file.pp

```diff
@@ -0,0 +1,17 @@
+# == Define: aerospike::xdr_credentials_file
+#
+define aerospike::xdr_credentials_file (
+  $all_xdr_credentials
+) {
+  if ! empty($all_xdr_credentials) {
+    $dc_credentials = $all_xdr_credentials[$name]
+    file { "/etc/aerospike/security-credentials_$name.txt":
+      ensure  => present,
+      content => template('aerospike/security-credentials.conf.erb'),
+      mode    => '0600',
+      owner   => $aerospike::system_user,
+      group   => $aerospike::system_group,
```

Comment:
Add those as default values of parameters. The defined should be independent of the class variables.
Doing that (and setting the parameters to the spec) should fix your failing tests issue.